internal/storage/udev_monitor.go

```diff
@@ -0,0 +1,100 @@
+package storage
+
+import (
+	"context"
+	"log"
+	"time"
+
+	udev "github.com/farjump/go-libudev"
+	"github.com/pkg/errors"
+)
+
+type DeviceStatus string
+
+const (
+	DeviceStatusActive   DeviceStatus = "active"
+	DeviceStatusInactive DeviceStatus = "inactive"
+	DeviceStatusUnknown  DeviceStatus = "unknown"
+	udevActionAdd                     = "add"
+	udevActionRemove                  = "remove"
+)
+
+type DeviceNum struct {
+	Major int
+	Minor int
+}
+
+type UdevDevice struct {
+	DeviceNode   string
+	DeviceStatus DeviceStatus
+}
+
+type UdevMonitor struct {
+	cancel  context.CancelFunc
+	ctx     context.Context
+	monitor *udev.Monitor
+	devices map[DeviceNum]UdevDevice
+}
+
+func NewUdevMonitor(ctx context.Context, cancel context.CancelFunc) (monitor *UdevMonitor) {
+	u := udev.Udev{}
+	m := u.NewMonitorFromNetlink("udev")
+	m.FilterAddMatchSubsystemDevtype("block", "disk")
+	ret := &UdevMonitor{
+		cancel:  cancel,
+		ctx:     ctx,
+		monitor: m,
+		devices: make(map[DeviceNum]UdevDevice),
```

Comment:
A nit:

I advise against using structs as map keys. Go allows us to use structs as map keys as long as all the fields can be compared for equality (which is the case here), but it's so easy to forget that you're using a struct as a map key and maybe add a slice to the struct later, breaking comparison.

You can use a string as a key here, composed of the major-minor like:

```go
mapKey := fmt.Sprintf("%d-%d", major, minor)
```

This would be internal to this package. You don't have to change the ```GetUdevDevice()``` signature.